django_lightweight_queue/worker.py

```diff
@@ -55,14 +56,23 @@ def run(self):
         backend = get_backend()
         self.log.info("Loaded backend %s", backend)
 
-        while self.running.value:
+        time_item_last_processed = datetime.datetime.now()
+
+        while self.running.value and not self.idle_time_reached(time_item_last_processed):
             try:
-                self.process(backend)
+                item_processed = self.process(backend)
+
+                if item_processed:
+                    time_item_last_processed = datetime.datetime.now()
+
             except KeyboardInterrupt:
                 sys.exit(1)
 
         self.log.info("Exiting")
 
+    def idle_time_reached(self, time_item_last_processed):
+        return datetime.datetime.now() - time_item_last_processed > datetime.timedelta(minutes=30)
```

Comment:
I wrote it like this, as I was not quite sure how to style it. I done something now 1816903